src/App.jsx

```diff
@@ -39,37 +40,37 @@ function App() {
         <MantineProvider withGlobalStyles withNormalizeCSS>
             <NotificationsProvider>
                 <UserProvider>
-                    {withNavBar(pathname) ? <NavigationBar width={navBarWidth} page={page}/> : ""}
-                    <Container
-                        fluid // Permet de s'assurer que le container prend la largeur maximale (largeur totale de la page)
-                        style={{
-                            margin: "0", //On s'assure qu'il n'y a aucune bordure de type margin ou padding
-                            padding: "0",
-                            // Si la navbar est affiché, on met du padding sur le Container pour éviter la superposition
-                            // de la nav bar sur le container
-                            marginLeft: (isSmallDevice || !withNavBar(pathname)) ? 0 : navBarWidth,
-                            paddingLeft: "0"
-                        }}>
-
-                        <Routes>
-                            <Route path="/">
-                                {/*public routes*/}
-                                <Route path="login" element={<Login/>}/>
-
-                                {/*privates routes*/}
-                                <Route element={<RequireAuth/>}>
-
-                                    <Route path="/" element={<Home setPage={setPage}/>}/>
-                                    <Route path="pg/:pgId" element={<PG setPage={setPage}/>}/>
-                                    <Route path="edit" element={<Edit setPage={setPage}/>}/>
-                                    <Route path="finss" element={<Finss setPage={setPage}/>}/>
-
+                    <CategorieProvider>
```

Comment:
t'es sûr que tu veux pas le mettre avec User ? car pour moi c'est vraiment une info qui irait bien avec les infos de l'utilisateur : l'utilisateur a sélectionné la catégorie bidule

Reply:
Ma pensée était la suivante:
Toutes les infos qu'on met en context comme à ce niveau ci sont vouées à être utiliséees partout dans le site.
On pourrait donc créer un seul context et mettre dedans Mantine theme, l'utilisateur, la catégorie sélectionnée, etc...

Mais ça ferait un context énorme et difficile à exploiter.
En découpant comme ça on garde des contextes plus petits, plus facile à comprendre et à utiliser. (lorsque on l'utilise on ne se retrouve pas avec une liste de n éléments dont on en utilise 1 seul)

Je pense que ce sont deux informations qui ne sont pas utilisées en même temps et qui ne devraient donc pas avoir le même context.

(remember: i can be wrong)